src/pages/AMM/Pools/components/DepositForm/DepositForm.tsx

```diff
@@ -21,84 +20,114 @@ import { getTokenPrice } from '@/utils/helpers/prices';
-    } catch (error: any) {
-      transaction.reject(error);
-    }
+        return { amounts, pool, slippage, deadline, accountId };
+      } catch (error: any) {
+        transaction.reject(error);
+      }
+    },
+    [accountId, pool, pooledCurrencies, slippage, transaction]
+  );
+
+  const handleSubmit = async (data: DepositLiquidityPoolFormData) => {
+    const transactionData = await getTransactionArgs(data);
+
+    if (!transactionData) return;
+
+    const { accountId, amounts, deadline, pool } = transactionData;
+
+    return transaction.execute(amounts, pool, slippage, deadline, accountId);
   };
 
-  const tokens = pooledCurrencies.reduce(
-    (acc: DepositLiquidityPoolValidationParams['tokens'], pooled) => ({
-      ...acc,
-      [pooled.currency.ticker]: {
-        minAmount: newMonetaryAmount(1, pooled.currency),
-        maxAmount: getAvailableBalance(pooled.currency.ticker) || newMonetaryAmount(0, pooled.currency)
-      }
-    }),
-    {}
+  const tokens = useMemo(
+    () =>
+      pooledCurrencies.reduce(
+        (acc: DepositLiquidityPoolValidationParams['tokens'], pooled) => ({
+          ...acc,
+          [pooled.currency.ticker]: {
+            minAmount: newMonetaryAmount(1, pooled.currency),
+            maxAmount: getAvailableBalance(pooled.currency.ticker) || newMonetaryAmount(0, pooled.currency)
+          }
+        }),
+        {}
+      ),
+    [getAvailableBalance, pooledCurrencies]
+  );
+
+  const defaultValues = pooledCurrencies.reduce((acc, amount) => ({ ...acc, [amount.currency.ticker]: '' }), {});
+
+  const initialValues = useMemo(
+    () => ({ ...defaultValues, [POOL_DEPOSIT_FEE_TOKEN_FIELD]: transaction.fee.defaultCurrency.ticker }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
   );
 
   const form = useForm<DepositLiquidityPoolFormData>({
-    initialValues: defaultValues,
-    validationSchema: depositLiquidityPoolSchema({ transactionFee: TRANSACTION_FEE_AMOUNT, governanceBalance, tokens }),
-    onSubmit: handleSubmit
-  });
+    initialValues,
+    validationSchema: depositLiquidityPoolSchema({ tokens }),
+    onSubmit: handleSubmit,
+    onComplete: async (values) => {
+      const transactionData = await getTransactionArgs(values);
 
-  const handleChange: ChangeEventHandler<HTMLInputElement> = (e) => {
-    if (isCustomAmountsMode(form)) return;
+      if (!transactionData) return;
 
-    if (!e.target.value || isNaN(Number(e.target.value))) {
-      return form.setValues(defaultValues);
+      const { accountId, amounts, deadline, pool } = transactionData;
+
+      const feeTicker = values[POOL_DEPOSIT_FEE_TOKEN_FIELD];
+
+      return transaction.fee.setCurrency(feeTicker).estimate(amounts, pool, slippage, deadline, accountId);
     }
+  });
 
+  const handleChange: ChangeEventHandler<HTMLInputElement> = (e) => {
     // If pool has no liquidity, the assets ratio is set by the user,
     // therefore the value inputted is directly used.
-    if (pool.isEmpty) {
```

Comment:
Removed this because it is not necessary. You should just return and exit the function.